app/controllers/concerns/period_selection.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
```

Comment:
CohortPeriod and Period are fundamentally different data. For example,
- The `CohortPeriod` period is the cohort’s duration. It can be a quarter/month.
- The `Period` is the query’s duration. It can be a quarter, month or day. 

This becomes more apparent when we switch pages between cohort reports and regular reports and want to preserve these filters.

Also, the concerns deal with defaulting each of them in a separate manner. It might be worth the overhead to keep them separate.